### DISCUSSION TITLE
Using tRPC

### DISCUSSION THREAD

**Discussion by hodeinavarro** [ykdojo/defaang#36] [Ideas]
I think the use of `tRPC` would benefit this project.

`tRPC` gives you typesafe APIs and also validates inputs on mutations, you can check them out [here](https://trpc.io) and [their github](https://github.com/trpc/trpc).

Version 10 is soon to be released.

**Bonus**: I'd create the project from `create-t3-app` and replace `prisma` for `supabase` if you end up going for it.

**Comment by ykdojo** [comment]
Interesting! Any way to use tRPC with Supabase?
